t/app/controller/report_inspect.t

```diff
@@ -594,6 +595,17 @@ FixMyStreet::override_config {
         is $report->comments->count, 1, "Only leaves one update";
         like $report->comments->first->text, qr/Category changed.*Badgers/, 'update text included category change';
     };
+
+    subtest "test non-public changing" => sub {
+        is $report->non_public, 0, 'Not set to non-public';
+        $mech->get_ok("/report/$report_id");
+        $mech->submit_form(button => 'save', with_fields => { include_update => 0, non_public => 1 });
+        $report->discard_changes;
+        is $report->non_public, 1, 'Now set to non-public';
+        $mech->submit_form(button => 'save', with_fields => { include_update => 0, non_public => 0 });
+        $report->discard_changes;
+        is $report->non_public, 0, 'Not set to non-public';
+    };
 };
```

Comment:
Possibly also a test to make sure it doesn't add an update just to avoid regressions.